day-four-challenge/secure-container/getPasswords.js

```diff
@@ -0,0 +1,33 @@
+const { map, sort, not, uniq, equals, unfold } = require('ramda');
+
+const MIN = 359282;
+const MAX = 820401;
+const PASSWORD_MIN_LENGTH = 6;
+
+const diff = (a, b) => { return a - b };
+
+const digitsIncreaseOrStayTheSameFromLeftToRight = array => (
+  equals(sort(diff, array), array)
+);
+
+const hasAtLeastTwoRepeatingDigits = (array) => (not(equals(uniq(array), array)))
+
+const isASixDigitNumber = (array) => array.length === PASSWORD_MIN_LENGTH;
+
+const meetsPasswordRequirements = (num) => {
+  const numStringArray = num.toString().split('');
+  const numArray = map(parseInt, numStringArray);
+
+  return isASixDigitNumber(numArray) && digitsIncreaseOrStayTheSameFromLeftToRight(numArray) && hasAtLeastTwoRepeatingDigits(numArray)
+}
```

Comment:
This is looking really solid, and I think I've found a solution for how to Ramdafy this. I'm not going to paste in the solution verbatim because I know that finding the solution yourself can be really rewarding, but reach out to me if you get stuck. In any case, let me set you on the right path.

Ramda has a really useful function for this case called, `allPass`, where you can pass in an array of predicates to evaluate a value. This matches up perfectly with what we're looking to do here. We want to effectively say, "here's an array: make sure it passes the following number of rules." This is really nice to use, especially if rules are added or removed down the line.

Well once we have our rules set up with `allPass`, let's add some more mad science! What if we broke out the function that creates our array into a different function? Then, we could use reduce `meetsPasswordRequirements` to a single function that uses `pipe` to generate an array based on a single number, and then send that value along to out `allPass` (either a separate function or an anon function: dealer's choice!) and make `meetsPasswordRequirements` so simple that all it's doing is 1) sending the number to a function to generate the array, and 2) sending it along to either an anon or dedicated function to evaluate business rules.

Let me know what you think! I have a working copy on my machine if you wanted some help along the way!

Reply:
This is an awesome strategy, thanks so much for the feedback! I've implemented it here: 09579fb707da9b4fb5183c6fd3a75a0af445244b. Let me know what you think!

Reply:
This is perfect, and was exactly the same approach I took! Well done! This looks 💯 !!!
![giphy](https://user-images.githubusercontent.com/7015466/74490883-f8d5a280-4e97-11ea-83a9-5213267737db.gif)